File: src/main/java/gg/mineral/server/entity/attribute/AttributeOperation.java
package gg.mineral.server.entity.attribute;

import org.jetbrains.annotations.Nullable;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
@Getter
public enum AttributeOperation {
    ADDITION(0),
    MULTIPLY_BASE(1),
    MULTIPLY_TOTAL(2);

    private static final AttributeOperation[] VALUES = new AttributeOperation[] { ADDITION, MULTIPLY_BASE,
            MULTIPLY_TOTAL };
    private final int id;

    @Nullable
    public static AttributeOperation fromId(int id) {
        if (id >= 0 && id < VALUES.length)
            return VALUES[id];

        return null;
    }
}

File: api/src/main/kotlin/gg/mineral/api/entity/attribute/AttributeOperation.kt
package gg.mineral.api.entity.attribute

enum class AttributeOperation(val id: Byte) {
    ADDITION(0),
    MULTIPLY_BASE(1),
    MULTIPLY_TOTAL(2);

    companion object {
        private val VALUES = arrayOf(
            ADDITION, MULTIPLY_BASE,
            MULTIPLY_TOTAL
        )

        fun fromId(id: Int): AttributeOperation? {
            if (id >= 0 && id < VALUES.size) return VALUES[id]

            return null
        }
    }
}